syntax = "proto3";

package printer;

message KeepAlive{
  bool keepAlive = 1;
}

message PrintConfig {
  int32 temperature = 1;
  string name = 2;
}

message ChangeConfig{

  enum ConfigChangeType{
    ADD = 0;
    REMOVE = 1;
  }

  ConfigChangeType changeType = 1;
  PrintConfig config = 2;
}

message RequestStatus{
  bool sendConfigs = 1;
}

message PrinterStatus{
  bool isTempControlActive = 1;
  float temperatureOutside = 2;
  float temperatureInsideTop = 3;
  float temperatureInsideBottom = 4;

  repeated PrintConfig printConfigs = 5;
}

